### PR TITLE
Cleanup unused env vars and rename for clarity

### PR DESCRIPTION
**Pull request by Fryguy**
Additionally, use the BASE_PATH env var as part of the API paths

@martinpovolny @Hyperkid123 Please review.  Let's also discuss before merge because I need to change the travis environment to have BASE_PATH set properly (and less importantly, so I can remove the rest of the env vars defined there).

### CODE REVIEW

File: src/Helpers/Shared/userLogin.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import { AdminsApi, ApiClient as ServicePortalApiClient } from '@manageiq/service-portal-api';
 import { DefaultApi, ApiClient as TopologicalInventoryApiClient } from '@manageiq/topological_inventory';
-import { SOURCES_API_BASE, SERVICE_PORTAL_API_BASE } from '../../Utilities/Constants';
+import { TOPOLOGICAL_INVENTORY_API_BASE, SERVICE_PORTAL_API_BASE } from '../../Utilities/Constants';
 
 const adminApi = new AdminsApi();
 
 const defaultClient = TopologicalInventoryApiClient.instance;
-defaultClient.basePath = SOURCES_API_BASE;
+defaultClient.basePath = TOPOLOGICAL_INVENTORY_API_BASE;
```
Comment (Fryguy) [inline src/Helpers/Shared/userLogin.js]: One weird thing here is that this defaultClient is not used in the getTopologicalUserApi method below, which is confusing.  Instead the method uses DefaultApi from '@manageiq/topological_inventory'. Are we setting a base path in something that isn't used?  Same thing for the sspDefaultClient

File: src/Helpers/Shared/userLogin.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import { AdminsApi, ApiClient as ServicePortalApiClient } from '@manageiq/service-portal-api';
 import { DefaultApi, ApiClient as TopologicalInventoryApiClient } from '@manageiq/topological_inventory';
```
Comment (Fryguy) [inline src/Helpers/Shared/userLogin.js]: Minor, but the package names are inconsistent.   I know we haven't moved over the service-portal-api client to the ManageIQ repo, but I expected the client package names to be consistent.

File: config/base.webpack.plugins.js [hunk 3]
```diff
@@ -87,14 +87,7 @@ plugins.push(CopyFilesWebpackPlugin);
...
 const envPlugin = new webpack.DefinePlugin({
-    'process.env.API_HOST': JSON.stringify(process.env.API_HOST),
-    'process.env.API_PORT': JSON.stringify(process.env.API_PORT),
-    'process.env.BASE_PATH': JSON.stringify(process.env.BASE_PATH),
-    'process.env.API_VERSION': JSON.stringify(process.env.API_VERSION),
-    'process.env.TOPO_API_HOST': JSON.stringify(process.env.TOPO_API_HOST),
-    'process.env.TOPO_API_PORT': JSON.stringify(process.env.TOPO_API_PORT),
-    'process.env.TOPO_BASE_PATH': JSON.stringify(process.env.TOPO_BASE_PATH),
-    'process.env.TOPO_API_VERSION': JSON.stringify(process.env.TOPO_API_VERSION)
+    'process.env.BASE_PATH': JSON.stringify(process.env.BASE_PATH) || '/r/insights/platform'
```
[the remark this answers is not in the document]
Comment (lgalis) [reply config/base.webpack.plugins.js]: i think there should be no default here, and instead have a default, if needed, where these variables are used.

File: src/Helpers/Shared/userLogin.js (hunk 1 above)
Comment (Hyperkid123) [reply src/Helpers/Shared/userLogin.js]: @Fryguy event though the ApiClients are not used directly, they are called byt adminsApi/usersApi. It is a singleton that should be used to configure your apps. That way you can change the basePaths and you don't have to hardcode them inside the clients.

File: src/Utilities/Constants.js [hunk 4]
```diff
@@ -1,3 +1,2 @@
-
-export const SOURCES_API_BASE = `/r/insights/platform/topological-inventory/v0.0`;
-export const SERVICE_PORTAL_API_BASE = `/r/insights/platform/service-portal/v0.0`;
+export const SERVICE_PORTAL_API_BASE = process.env.BASE_PATH + '/service-portal/v0.0';
+export const TOPOLOGICAL_INVENTORY_API_BASE = process.env.BASE_PATH + '/topological-inventory/v0.0';
```
Comment (Hyperkid123) [inline src/Utilities/Constants.js]: i think the '/r' path is required. It marks the path as an API.

File: src/Helpers/Shared/userLogin.js (hunk 2 above)
Comment (Hyperkid123) [reply src/Helpers/Shared/userLogin.js]: @Fryguy i agree

File: src/Utilities/Constants.js (hunk 4 above)
Comment (lgalis) [inline src/Utilities/Constants.js]: Should we have environment variables for /service-portal/v0.0 and  /topological-inventory/v0.0?

**Comment by Fryguy** [comment]
@Hyperkid123 were you able to get back to this?

File: src/Utilities/Constants.js (hunk 4 above)
Comment (Fryguy) [reply src/Utilities/Constants.js]: Maybe?  It's probably smart to have that and default it when not set.

File: config/base.webpack.plugins.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (lgalis) [reply config/base.webpack.plugins.js]: I would take the default out from here and move it to where the BASE_PATH is used.

File: src/Utilities/Constants.js (hunk 4 above)
Comment (lgalis) [reply src/Utilities/Constants.js]: This should be:

onst base_path = process.env.BASE_PATH || '/r/insights/platform';

export const SERVICE_PORTAL_API_BASE = `${base_path}/service-portal/v0.0`;
export const TOPOLOGICAL_INVENTORY_API_BASE = `${base_path}/topological-inventory/v0.0`;

**Comment by Hyperkid123** [comment]
@Fryguy OK se the issue is was very simple. The default basePath must be stringyfied otherwise it is not a string but a regular expression. Can be fixed at `base.webpack.plugins` with this small patch:

```diff
diff --git a/config/base.webpack.plugins.js b/config/base.webpack.plugins.js
index 3c822a1..91cc7f9 100644
--- a/config/base.webpack.plugins.js
+++ b/config/base.webpack.plugins.js
@@ -87,7 +87,7 @@ plugins.push(CopyFilesWebpackPlugin);
  * Makes build-time env vars available to the client-side as constants
  */
 const envPlugin = new webpack.DefinePlugin({
-    'process.env.BASE_PATH': JSON.stringify(process.env.BASE_PATH) || '/r/insights/platform'
+    'process.env.BASE_PATH': JSON.stringify(process.env.BASE_PATH) || JSON.stringify('/r/insights/platform')
 });
 plugins.push(envPlugin);
```

**Comment by Hyperkid123** [comment]
Apart from the small issue it looks good. After the patch API still works. I've got just a small nitpicks.

File: src/Utilities/Constants.js [hunk 5]
```diff
@@ -1,3 +1,2 @@
-
-export const SOURCES_API_BASE = `/r/insights/platform/topological-inventory/v0.0`;
-export const SERVICE_PORTAL_API_BASE = `/r/insights/platform/service-portal/v0.0`;
+export const SERVICE_PORTAL_API_BASE = process.env.BASE_PATH + '/service-portal/v0.0';
```
Comment (Hyperkid123) [inline src/Utilities/Constants.js]: Can you please use template strings. Its just about code styling.
```javascript
export const SERVICE_PORTAL_API_BASE = `${process.env.BASE_PATH}/service-portal/v0.0`;
```

File: src/Utilities/Constants.js (hunk 4 above)
Comment (Fryguy) [reply src/Utilities/Constants.js]: I meant to put the default inside the parens...just a bug.

File: src/Utilities/Constants.js (hunk 5 above)
Comment (Fryguy) [reply src/Utilities/Constants.js]: I really don't understand template strings.  Can you explain why they are preferable over normal strings or give me a link with a good explanation?

**Comment by Fryguy** [comment]
> The default basePath must be stringyfied otherwise it is not a string but a regular expression. Can be fixed at `base.webpack.plugins` with this small patch:

Ah yeah, I meant to put the || inside the parens... see https://github.com/ManageIQ/service_portal-ui/pull/80#discussion_r253582126

File: src/Utilities/Constants.js (hunk 5 above)
Comment (Hyperkid123) [reply src/Utilities/Constants.js]: @Fryguy for this case its just a different syntax. But it has some neat features especially for string interpolation. Also the `+` is ambiguous and in some cases may have bad side effects (if you don't have 100% control over the variable content). That won't happen in this case obviously but we are using it across the code and i would like to keep it consistent.

There is a lot of articles across the internet. But this one is good imo: https://developers.google.com/web/updates/2015/01/ES6-Template-Strings

**Pull request merged**